server/build.gradle

```diff
@@ -117,44 +118,55 @@ compileJava {
 
 // Creates the full jar file, with the extender server classes included
 bootJar {
-    baseName = 'extender'
+    archiveBaseName = 'extender'
     version =  '0.1.0'
 }
 
 // Builds the extender server base classes into a .jar file
 jar {
-    baseName = 'extender-base'
+    archiveBaseName = 'extender-base'
     version =  '0.1.0'
     archiveClassifier = '' // use empty string to avoid the 'plain' suffix
 }
 
-task buildDocker(type: Docker) {
-    dependsOn 'build', ':manifestmergetool:mainJar'
-    push = false
-    applicationName = bootJar.baseName
-    dockerfile = file('docker/Dockerfile')
+File dockerBuildDir = new File(project.buildDir, 'docker-stage');
+
+task dockerStageDir {
     doFirst {
+        if (dockerBuildDir.exists()) {
+            dockerBuildDir.delete();
+        }
+        dockerBuildDir.mkdirs();
         copy {
             from(project.projectDir) {
                 include "users/**"
             }
-            into stageDir
+            into dockerBuildDir
         }
         copy {
             from bootJar
-            into stageDir
+            into dockerBuildDir
         }
         copy {
             from project(':manifestmergetool').collect { it.tasks.withType(Jar) }
-            into stageDir
+            into dockerBuildDir
         }
         copy {
             from 'docker'
-            into stageDir
+            into dockerBuildDir
         }
     }
 }
 
+task buildDocker(type: DockerBuildImage) {
+    dependsOn 'build', ':manifestmergetool:mainJar', 'dockerStageDir'
+    inputDir = dockerBuildDir
+    imageId = bootJar.archiveBaseName
+    dockerFile = new File(dockerBuildDir, 'Dockerfile')
+    images.add("extender/extender:latest")
+    images.add("extender/extender:" + project.outSha)
```

Comment:
Produce images with different tags. It helps to identify builds from different commit.